lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -296,7 +296,8 @@
     - ' - C(dns_suffix) (list): List of domain suffixes, also known as DNS search path (default: C(domain) parameter).'
     - ' - C(domain) (string): DNS domain name to use.'
     - ' - C(hostname) (string): Computer hostname (default: shorted C(name) parameter). Allowed characters are alphanumeric (uppercase and lowercase)
-          and minus, rest of the characters are dropped as per RFC 952.'
+          and minus, rest of the characters are dropped as per RFC 952. Ansible 2.8 and onwards, this is a required parameter.
```

Comment:
I suggest a rephrasing here, because:
- Even in the old approach (force=True) you are not actually dropping individual characters, and that's what it sounds to me.
- Citing RFC 952 does not seem to me to be of much help to the module user, IMHO.
```suggestion
          and minus (no dots allowed, VMware guest customization for linux does not support receiving a FQDN name). From Ansible 2.8 onwards, this is a required parameter.
```

Reply:
Oh sorry, I see now that the code DOES drops individual characters in the code line just below your changes. Which is goes back to my original point. I think that is misleading. See the code I proposed in a comment.